### PR TITLE
Wdio JUnit Reporter - fix #5637

### PR DESCRIPTION
**Pull request by unickq**
## Fix - #5637

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

https://github.com/webdriverio/webdriverio/blob/915f1cf8d8b0b0312386b73fe471531b1736b4c3/packages/wdio-junit-reporter/src/index.js#L176

`runner` may not have those properties - check the https://github.com/webdriverio/webdriverio/pull/5637#issuecomment-661073765

### CODE REVIEW

File: packages/wdio-junit-reporter/src/index.js [hunk 1]
```diff
@@ -172,16 +172,21 @@ class JunitReporter extends WDIOReporter {
...
-                .map((capability) => capability.toLowerCase())
-                .join('.')
-                .replace(/ /g, '')
-
+            let browserstackSanitizedCapabilities;
+            try {
+                browserstackSanitizedCapabilities = [
+                    runner.capabilities.device,
+                    runner.capabilities.os,
+                    runner.capabilities.os_version.replace(/\./g, "_"),
```
Comment (christian-bromann) [inline packages/wdio-junit-reporter/src/index.js:180]:
```suggestion
                    (runner.capabilities.os_version || '').replace(/\./g, "_"),
```

Comment (unickq) [reply packages/wdio-junit-reporter/src/index.js]: done

File: packages/wdio-junit-reporter/src/index.js [hunk 2]
```diff
@@ -172,16 +172,18 @@ class JunitReporter extends WDIOReporter {
         if (runner.config.hostname !== undefined && runner.config.hostname.indexOf('browserstack') > -1) {
             // NOTE: deviceUUID is used to build sanitizedCapabilities resulting in a ever-changing package name in runner.sanitizedCapabilities when running Android tests under Browserstack. (i.e. ht79v1a03938.android.9)
             // NOTE: platformVersion is used to build sanitizedCapabilities which can be incorrect and includes a minor version for iOS which is not guaranteed to be the same under Browserstack.
-            const browserstackSanitizedCapabilities = [
+            let browserstackSanitizedCapabilities = [
```
Comment (christian-bromann) [inline packages/wdio-junit-reporter/src/index.js:175]:
```suggestion
            const browserstackSanitizedCapabilities = [
```

File: packages/wdio-junit-reporter/src/index.js [hunk 3]
```diff
@@ -172,16 +172,18 @@ class JunitReporter extends WDIOReporter {
...
                 runner.capabilities.device,
                 runner.capabilities.os,
-                runner.capabilities.os_version.replace(/\./g, '_')
+                (runner.capabilities.os_version || "").replace(/\./g, "_"),
```
Comment (christian-bromann) [inline packages/wdio-junit-reporter/src/index.js:178]:
```suggestion
                (runner.capabilities.os_version || '').replace(/\./g, '_'),
```

File: packages/wdio-junit-reporter/src/index.js [hunk 4]
```diff
@@ -172,16 +172,18 @@ class JunitReporter extends WDIOReporter {
...
                 .filter(Boolean)
                 .map((capability) => capability.toLowerCase())
-                .join('.')
-                .replace(/ /g, '')
-
+                .join(".")
+                .replace(/ /g, "");
+            if (!browserstackSanitizedCapabilities) {
+                browserstackSanitizedCapabilities = runner.sanitizedCapabilities;
+            }
```
Comment (christian-bromann) [inline packages/wdio-junit-reporter/src/index.js:186]:
```suggestion
                .replace(/ /g, "") || runner.sanitizedCapabilities;
```

**Pull request merged**